### ISSUE TITLE
Install `git` inside core24 build environments

### ISSUE THREAD

**Issue by mr-cal** [snapcore/snapcraft#4944]
### What needs to get done

1. Update [`snapcraft.services.provider.setup()`](https://github.com/canonical/snapcraft/blob/4ee3c1493aec801c9b1e6737e51c877b3e03dc79/snapcraft/services/provider.py#L28) to install `git`:
```diff
diff --git a/snapcraft/services/provider.py b/snapcraft/services/provider.py
index 2ed2b590..e28b730c 100644
@@ -26,6 +26,12 @@ class Provider(ProviderService):
 
     @overrides
     def setup(self) -> None:
+        """Snapcraft-specific setup."""
+        # required for apt gpg and `version: git`
+        self.packages.extend(["gpg", "dirmngr", "git"])
+
         if build_info := os.getenv("SNAPCRAFT_BUILD_INFO"):
             self.environment["SNAPCRAFT_BUILD_INFO"] = build_info
         if image_info := os.getenv("SNAPCRAFT_IMAGE_INFO"):
```

2. Update the compatibility tag in snapcraft (blocked by https://github.com/canonical/craft-application/issues/105)

### Why it needs to get done

- To allow `version: git` and other `git` calls to work inside the build environment
- [Parity with core22](https://github.com/canonical/snapcraft/blob/4ee3c1493aec801c9b1e6737e51c877b3e03dc79/snapcraft/providers.py#L210)

**Comment by mr-cal** [comment]
This should be fixed because we bundle git in #5149

**Comment by farshidtz** [comment]
I can still reproduce this:

```console
$ snap list snapcraft 
Name       Version                  Rev    Tracking     Publisher   Notes
snapcraft  8.5.0.post1+gitc7d11ac6  13023  latest/edge  canonical✓  classi

$ snapcraft init
Go to https://docs.snapcraft.io/the-snapcraft-format/8337 for more information about the snapcraft.yaml format.
Successfully initialised project.

$ cat snap/snapcraft.yaml | yq .base
core24

$ nano snap/snapcraft.yaml # set version to 'git'

$ git init
Initialized empty Git repository in /tmp/snaptest/.git/

$ git commit -m "init"
[main (root-commit) 43545df] init
 1 file changed, 17 insertions(+)
 create mode 100644 snap/snapcraft.yaml

$ git tag test

$ snapcraft 
snapcraft internal error: FileNotFoundError(2, 'No such file or directory')
Failed to execute snapcraft in instance.
Full execution log: '/home/farshid/.local/state/snapcraft/log/snapcraft-20241122-113431.080823.log'
# thanks for nothing

$ snapcraft -v
Starting snapcraft, version 8.5.0.post1+gitc7d11ac6
Logging execution to '/home/farshid/.local/state/snapcraft/log/snapcraft-20241122-113309.982184.log'
Launching managed ubuntu 24.04 instance...
Starting instance 
Starting snapcraft, version 8.5.0.post1+gitc7d11ac6
Logging execution to '/tmp/snapcraft.log'
Initialising lifecycle
Installing build-packages
Installing build-snaps
Skipping pull for my-part (already ran)
Skipping build for my-part (already ran)
Skipping stage for my-part (already ran)
Skipping prime for my-part (already ran)
Determining the version from the project repo (version: git).
snapcraft internal error: FileNotFoundError(2, 'No such file or directory')
Failed to execute snapcraft in instance.
Full execution log: '/home/farshid/.local/state/snapcraft/log/snapcraft-20241122-113309.982184.log'
```

Can be fixed by adding `git` as a build package:
```yaml
parts:
  my-part:
    # See 'snapcraft plugins'
    plugin: nil
    build-packages:
      - git
```
